src/inventory/inventory.resolver.ts

```diff
@@ -19,24 +27,61 @@ export class InventoryResolver {
     );
   }
 
-  @Query(() => [ProductItem], { name: 'productItems' })
-  findAll() {
-    return this.inventoryService.findAll();
+  @Query(() => ProductItemConnection, {
+    name: 'productItems',
+    description: 'Retrieves all product items',
+  })
+  async findAll(
+    @Args() args: FindProductItemArgs,
+  ): Promise<IPaginatedType<ProductItem>> {
+    if (!args.orderBy) {
+      args.orderBy = {
+        field: ProductItemOrderField.ID,
+        direction: 1,
+      };
+    }
+    const [nodes, totalCount] = await this.inventoryService.findAll(args);
```

Comment:
totalCount and nodes should be resolved independantly to prevent performance issues when e.g. only the totalCount is requested by the query